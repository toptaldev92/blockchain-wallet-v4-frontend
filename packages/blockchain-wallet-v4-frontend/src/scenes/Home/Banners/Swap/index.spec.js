import { shallow } from 'enzyme'
import { SwapBanner } from './index'
import React from 'react'
import toJson from 'enzyme-to-json'

jest.mock('blockchain-info-components', () => ({
  Button: 'button',
  Text: 'text',
  Image: 'image'
}))

describe('SwapBanner', () => {
  it('renders null w/o showBanner', () => {
    const component = shallow(<SwapBanner />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with getstarted button', () => {
    const component = shallow(<SwapBanner showBanner kycNotFinished />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with exchange link', () => {
    const component = shallow(<SwapBanner showBanner />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
