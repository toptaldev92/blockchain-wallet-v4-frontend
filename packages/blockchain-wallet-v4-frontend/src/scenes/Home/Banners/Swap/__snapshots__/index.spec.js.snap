// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapBanner renders correctly with exchange link 1`] = `
<Swap__Wrapper>
  <Swap__Column>
    <Swap__LargeText>
      <FormattedMessage
        defaultMessage="Swap Your Crypto"
        id="scenes.home.banners.swap.title"
        values={Object {}}
      />
    </Swap__LargeText>
    <Swap__MediumText>
      <FormattedMessage
        defaultMessage="Trading your crypto doesn't mean trading away control."
        id="scenes.home.banners.swap.subtitle"
        values={Object {}}
      />
    </Swap__MediumText>
  </Swap__Column>
  <Swap__Column
    hiddenOnMobile={true}
  >
    <Swap__BackgroundImage
      name="swap-dashboard-right"
    />
  </Swap__Column>
  <Swap__Column>
    <LinkContainer
      activeClassName="active"
      exact={false}
      replace={false}
      strict={false}
      to="/swap"
    >
      <Swap__GetStartedButton>
        <FormattedMessage
          defaultMessage="Make Swap"
          id="scenes.home.banners.swap.makeswap"
          values={Object {}}
        />
      </Swap__GetStartedButton>
    </LinkContainer>
  </Swap__Column>
</Swap__Wrapper>
`;

exports[`SwapBanner renders correctly with getstarted button 1`] = `
<Swap__Wrapper>
  <Swap__Column>
    <Swap__LargeText>
      <FormattedMessage
        defaultMessage="Swap Your Crypto"
        id="scenes.home.banners.swap.title"
        values={Object {}}
      />
    </Swap__LargeText>
    <Swap__MediumText>
      <FormattedMessage
        defaultMessage="Trading your crypto doesn't mean trading away control."
        id="scenes.home.banners.swap.subtitle"
        values={Object {}}
      />
    </Swap__MediumText>
  </Swap__Column>
  <Swap__Column
    hiddenOnMobile={true}
  >
    <Swap__BackgroundImage
      name="swap-dashboard-right"
    />
  </Swap__Column>
  <Swap__Column>
    <Swap__GetStartedButton>
      <FormattedMessage
        defaultMessage="Get Started"
        id="scenes.home.banners.swap.started"
        values={Object {}}
      />
    </Swap__GetStartedButton>
  </Swap__Column>
</Swap__Wrapper>
`;

exports[`SwapBanner renders null w/o showBanner 1`] = `
<Swap__Wrapper>
  <Swap__Column>
    <Swap__LargeText>
      <FormattedMessage
        defaultMessage="Swap Your Crypto"
        id="scenes.home.banners.swap.title"
        values={Object {}}
      />
    </Swap__LargeText>
    <Swap__MediumText>
      <FormattedMessage
        defaultMessage="Trading your crypto doesn't mean trading away control."
        id="scenes.home.banners.swap.subtitle"
        values={Object {}}
      />
    </Swap__MediumText>
  </Swap__Column>
  <Swap__Column
    hiddenOnMobile={true}
  >
    <Swap__BackgroundImage
      name="swap-dashboard-right"
    />
  </Swap__Column>
  <Swap__Column>
    <LinkContainer
      activeClassName="active"
      exact={false}
      replace={false}
      strict={false}
      to="/swap"
    >
      <Swap__GetStartedButton>
        <FormattedMessage
          defaultMessage="Make Swap"
          id="scenes.home.banners.swap.makeswap"
          values={Object {}}
        />
      </Swap__GetStartedButton>
    </LinkContainer>
  </Swap__Column>
</Swap__Wrapper>
`;
